201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json

```diff
@@ -0,0 +1,511 @@
+        },
+        "app-gateway-capacity": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "The number of Azure Application Gateway capacity units to provision. This setting has a direct impact on consumption cost and is recommended to be left at the default value of 1"
+            }
+        },
+        "vnet-address-prefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "The address space (in CIDR notation) to use for the VNET to be deployed in this solution. If integrating with other networked components, there should be no overlap in address space."
+            }
+        },
+        "app-gateway-subnet-prefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/24",
+            "metadata": {
+                "description": "The address space (in CIDR notation) to use for the subnet to be used by Azure Application Gateway. Must be contained in the VNET address space."
+            }
+        },
+        "apim-subnet-prefix": {
+            "type": "string",
+            "defaultValue": "10.0.1.0/24",
+            "metadata": {
+                "description": "The address space (in CIDR notation) to use for the subnet to be used by Azure API Management. Must be contained in the VNET address space."
+            }
+        },        
+        "apim-publisher-name": {
+            "type": "string",
+            "defaultValue": "Contoso",
+            "metadata": {
+                "description": "Descriptive name for publisher to be used in the portal"
+            }
+        },
+        "apim-publisher-email": {
+            "type": "string",
+            "defaultValue": "api@contoso.com",
+            "metadata": {
+                "description": "Email adddress associated with publisher"
+            }
+        }
+    },
+    "variables": {
+        "app-insights-name": "[concat(parameters('base-name'), '-ai')]",
+        "vnet-name": "[concat(parameters('base-name'), '-vnet')]",
+        "apim-name": "[concat(parameters('base-name'), '-apim')]",
+        "public-ip-name": "[concat(parameters('base-name'), '-pip')]",
+        "app-gateway-name": "[concat(parameters('base-name'), '-agw')]",
+        "vnet-dns-link-name": "[concat(parameters('base-name'), '-vnet-dns-link')]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.ApiManagement/service",
+            "apiVersion": "2020-12-01",
+            "name": "[variables('apim-name')]",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "[parameters('apim-sku')]",
+                "capacity": "[parameters('apim-capacity')]"
+            },
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "resources": [
+                {
+                    "type": "Microsoft.ApiManagement/service/gateways",
```

Comment:
since this is nested, just use "gateways" as the type

Reply:
Done - really helps readability